### PR TITLE
chore: bump operator rs 0.67

### PR DESCRIPTION
**Pull request by fhennig**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Review by soenkeliebau — APPROVED** [review]
Went over the changes, and tested it in a local kind cluster, looking good to me.

One thing I noticed was that I got clippy lints when running clippy locally, but we decided to not do anything about these for now (opened https://github.com/stackabletech/product-config/issues/107), and CI didn't find them for some reason 
 - opened https://github.com/stackabletech/nifi-operator/issues/617 for this

**Comment by sbernauer** [comment]
When you open up the files in your IDE they show up as yellow, I would be very happy if we would put the big variants behind a `Box` (also should improve the performance)
I don't want to dictate the effort on you, I already have a patch below (it also fixes a typo) :)
```
diff --git a/rust/operator-binary/src/controller.rs b/rust/operator-binary/src/controller.rs
index 808d69a..de1f580 100644
--- a/rust/operator-binary/src/controller.rs
+++ b/rust/operator-binary/src/controller.rs
@@ -182,8 +182,11 @@ pub enum Error {
         source: stackable_operator::builder::meta::Error,
     },
 
-    #[snafu(display("Failed to load Product Config"))]
-    ProductConfigLoadFailed { source: config::Error },
+    #[snafu(display("Failed to load Product config"))]
+    ProductConfigLoadFailed {
+        #[snafu(source(from(config::Error, Box::new)))]
+        source: Box<config::Error>,
+    },
 
     #[snafu(display("Failed to find information about file [{}] in product config", kind))]
     ProductConfigKindNotSpecified { kind: String },
@@ -207,11 +210,15 @@ pub enum Error {
     NoRoleServiceFqdn,
 
     #[snafu(display("Bootstrap configuration error"))]
-    BoostrapConfig { source: crate::config::Error },
+    BootstrapConfig {
+        #[snafu(source(from(config::Error, Box::new)))]
+        source: Box<config::Error>,
+    },
 
     #[snafu(display("failed to prepare NiFi configuration for rolegroup {rolegroup}"))]
     BuildProductConfig {
-        source: crate::config::Error,
+        #[snafu(source(from(config::Error, Box::new)))]
+        source: Box<config::Error>,
         rolegroup: RoleGroupRef<NifiCluster>,
     },
 
@@ -731,7 +738,7 @@ async fn build_node_rolegroup_config_map(
                     })?
                     .clone(),
             )
-            .context(BoostrapConfigSnafu)?,
+            .context(BootstrapConfigSnafu)?,
         )
         .add_data(
             NIFI_PROPERTIES,
```

File: Cargo.toml [hunk 1]
```diff
@@ -23,10 +23,10 @@ semver = "1.0"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 serde_yaml = "0.9"
-snafu = "0.7"
-stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.64.0" }
-strum = { version = "0.25", features = ["derive"] }
-tokio = { version = "1.29", features = ["full"] }
+snafu = "0.8"
+stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "stackable-operator-0.67.0" }
```
Comment (maltesander) [inline Cargo.toml:27]: Clap can be updated to 4.5.
```suggestion
stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "stackable-operator-0.67.1" }
```

File: rust/crd/src/affinity.rs [hunk 2]
```diff
@@ -83,6 +83,8 @@ mod tests {
                                         )
                                     ]))
                                 }),
+                                match_label_keys: None,
+                                mismatch_label_keys: None,
                                 namespace_selector: None,
                                 namespaces: None,
```
Comment (maltesander) [inline rust/crd/src/affinity.rs:89]: can be defaulted instead.

File: rust/crd/src/lib.rs [hunk 3]
```diff
@@ -24,7 +24,7 @@ use stackable_operator::{
     k8s_openapi::{api::core::v1::Volume, apimachinery::pkg::api::resource::Quantity},
     kube::{runtime::reflector::ObjectRef, CustomResource, ResourceExt},
     memory::{BinaryMultiple, MemoryQuantity},
-    product_config_utils::{ConfigError, Configuration},
+    product_config_utils::{Configuration, Error as ConfigError},
```
Comment (maltesander) [inline rust/crd/src/lib.rs:27]: We now went without renaming the error in other operators.

Comment (fhennig) [reply rust/crd/src/lib.rs:27]: What do you mean here? The error is not renamed, right? do I need to change anything?

Comment (maltesander) [reply rust/crd/src/lib.rs:27]: We use the "full qualified name" in other operators, its just for consistency e.g. like https://github.com/stackabletech/zookeeper-operator/blob/4b42039afc67a87d64b12860ed0d146dfa3cfcaa/rust/crd/src/lib.rs#L330

**Review by maltesander — APPROVED** [review]
Thanks, LGTM!

**Review by sbernauer — APPROVED** [review]
Thanks

**Pull request merged**